packages/xml-toolkit/lib/extension.js

```diff
@@ -16,21 +17,37 @@ function activate() {
     },
   };
 
+  const meta = require(resolve(__dirname, "..", "package.json"));
+  const outChannel = window.createOutputChannel(meta.displayName);
   let clientOptions = {
     documentSelector: [{ scheme: "file", language: "xml" }],
     synchronize: {
       fileEvents: workspace.createFileSystemWatcher("**/*.xml"),
     },
+    outputChannel: outChannel,
   };
 
   client = new LanguageClient(
-    "XMLforVSCode",
-    "XML For VSCode",
+    meta.name,
+    meta.displayName,
     serverOptions,
     clientOptions
   );
 
+  // https://github.com/microsoft/vscode-languageserver-node/blob/bb85e80617c7f0cb64dad9dad852219e2da7c4c9/jsonrpc/src/node/test/connection.test.ts#L484-L507
   client.start();
+  initSWAInLangServer(client, meta);
+}
+
+/**
+ * @param {BaseLanguageClient} client
+ * @param {object} meta
+ * @returns {Promise<void>}
+ */
+async function initSWAInLangServer(client, meta) {
+  await client.onReady();
+  // Due to limitations in the SWA library, the usage analytics does not work currently on the VSCode engine
+  client.sendRequest("initSWA", meta.publisher, meta.name);
```

Comment:
You can transfer custom fields when initializing the server. Maybe it would be better than using a custom request, unless you want to call this more than once.
See client implementation:
https://github.com/SAP/ui5-language-assistant/blob/master/packages/vscode-ui5-language-assistant/src/extension.ts#L43 
and server implementation:
https://github.com/SAP/ui5-language-assistant/blob/master/packages/language-server/src/server.ts#L62

Reply:
I think the initialization params you suggested are better as we avoid possible race conditions
Where we want to Track/Log something yet the SWA/Logger was not yet initialized.

However we may still need the `sendRequest`, e.g to support dynamic changes (e.g logging scenario)